src/sandstorm/activity.capnp

```diff
@@ -0,0 +1,180 @@
+# Sandstorm - Personal Cloud Sandbox
+# Copyright (c) 2016 Sandstorm Development Group, Inc. and contributors
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+@0xa4e001d4cbcf33fa;
+
+$import "/capnp/c++.capnp".namespace("sandstorm");
+
+using Util = import "util.capnp";
+
+struct ActivityEvent {
+  # Describes an event in a grain's activity feed / log.
+  #
+  # Call SessionContext.activity() to post new activity. The activity is attributed to the user
+  # whose SessionContext was used.
+
+  path @0 :Text;
+  # Path (as in, URL) within the grain where the user can see this activity. Users inspecting the
+  # event will be linked to this location. Visiting this path implicitly marks the event as "read".
+  #
+  # This should NOT include a leading '/'. (So, an empty string -- the default -- goes to the grain
+  # root).
+  #
+  # Users are able to subscribe to events at specific paths to get notified on activity.
```

Comment:
I'm a little skeptical about this string-based query language for subscription paths and filtering, and wondering if something more explicitly typed would have clearer semantics.  For instance, it's not clear to me if I'm subscribed to `"foo?bar=baz"` and an event posts to `"foo?quux=grault&bar=baz"` whether I should expect to receive that event.  Do query strings represent an unordered map of filters, an ordered map of filters, or are they being treated as string-prefix comparisons?

Another risk stemming from the use of the actual URL path here is that an app wants to change internal URLs for something that previously it would have rendered the UI for itself - now the app cannot ever change these paths, even if they fix up all internal links, without breaking notifications.  This is probably okay in most cases, but larger projects may resist this sort of requirement.